chia/wallet/util/compute_hints.py

```diff
@@ -1,22 +1,33 @@
 from __future__ import annotations
 
-from typing import List
+from typing import Dict, Tuple
 
-from chia.types.blockchain_format.program import INFINITE_COST
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import INFINITE_COST, Program
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.coin_spend import CoinSpend
 from chia.types.condition_opcodes import ConditionOpcode
+from chia.util.ints import uint64
 
 
-def compute_coin_hints(cs: CoinSpend) -> List[bytes32]:
+def compute_coin_hints(cs: CoinSpend) -> Tuple[Dict[bytes32, bytes32], Dict[bytes32, Coin]]:
```

Comment:
While this is true of the places we currently use it, I don't think that's the intention of this as a utility function.  I'd like to preserve its flexibility since it's already there.

Reply:
I don't know, this doesn't look like a good change to me tbh.. and we probably shouldn't so much care about potential future use cases here which at the end might never be used/useful. Now you:
- Add more complexity to a function which already has more than it needs for a potential future use case which may or may never be used.
- Add complexity to the call sites
- Return a tuple which is something better to avoid in general
- For the returned tuple you even only in one place you use the second returned value and then you only use it to filter out the value you are concerned about. 

If you want to do something for the future you can maybe instead do something like

```python
@dataclass
class CoinHint:
  coin_id: bytes32
  hint: bytes32

def coin_hint_from_condition(condition: Program) -> CoinHint:
  ...

def coin_hint_for_coin_id(coin_spend: CoinSpend, coin_id: bytes32) -> CoinHint:
  ...
```

and write unit tests for both. Then if someone really comes up with some use case to get all hints from a spend they can simply reuse your tested `coin_hint_from_condition` and do a easily do a tested:

```python
def coin_hints(coin_spend: CoinSpend) -> List[CoinHint]:
  ...
```

Reply:
> Add more complexity to a function
I don't see how I added much complexity if at all for returning hints in a dictionary rather than in a list,  The core loop is the same (reworked a smidge to make it more robust, but semantically the same).

> Add complexity to the call sites
I think all of the call sites got significantly less complicated because they could just lookup the value they wanted instead of iterating through a list until they find the right one.

> Return a tuple which is something better to avoid in general
I wasn't aware that this was bad.  There's lots of places we return tuples, and I think it makes sense here (although maybe the function could use a rename to better identify that it's going to return a tuple.

> For the returned tuple you even only in one place you use the second returned value and then you only use it to filter out the value you are concerned about.

That's the point of the second value lol.  If there was such thing as a dictionary in python where each item had three values and you could look up any of the values by either of the other two, I'd be returning that.  Alas, I'm stuck with using two dictionaries for this functionality, but that's _why_ I'm returning both.  So you can filter out the coins you want and then easily look up the hints from them.

Reply:
> I don't see how I added much complexity if at all for returning hints in a dictionary rather than in a list, The core loop is the same (reworked a smidge to make it more robust, but semantically the same).

> I think all of the call sites got significantly less complicated because they could just lookup the value they wanted instead of iterating through a list until they find the right one.

Sure its not a lot more complexity added, but still some, while the variation i proposed reduces it. I thought instead of having more back and forth discussion here im just going to implement my thoughts in #15274, maybe you see what i mean there. It also avoid pre-computing all hints by just yielding the result after each. 

> There's lots of places we return tuples, and I think it makes sense here (although maybe the function could use a rename to better identify that it's going to return a tuple.

There are a lot other things which are not good in our code which doesn't mean we should continue to do this things. In general the tuple returns just make things confusing and messy for changes in the return value imo. And you end up with unused variables or a bunch of `_, the_value_i_want, _` constructs. Instead you can just introduce a `dataclass` for it which makes things more obvious and easier to adjust later.  

> That's the point of the second value lol. If there was such thing as a dictionary in python where each item had three values and you could look up any of the values by either of the other two, I'd be returning that.

I guess you mean where each item "had two values", which is basically 
```python
@dataclass
class TwoValues:
    one: int
    two: int

d: Dict[bytes32, TwoValues]
```

> I'm stuck with using two dictionaries for this functionality, but that's _why_ I'm returning both. So you can filter out the coins you want and then easily look up the hints from them.

Well since you anyway already used a `Tuple` you could have just returned one dict like `Dict[bytes32, Tuple[Coin, bytes32]]` but still, just adding the `dataclass` and return `Dict[bytes32, HintedCoin]` is the option we should prefer here still imo.